hwilib/descriptor.py

```diff
@@ -68,6 +68,16 @@ def __init__(
         self.pubkey = pubkey
         self.deriv_path = deriv_path
 
+        # Make ExtendedKey from pubkey if it isn't hex
+        self.extkey = None
+        try:
+            unhexlify(self.pubkey)
+            # Is hex, normal pubkey
+        except Exception:
+            # Not hex, maybe xpub
+            if self.pubkey[0:4] == "xpub" or self.pubkey[0:4] == "tpub":
```

Comment:
`if self.pubkey[0:4] in ["xpub", "tpub"]:`

Reply:
It would be better to move the check inside of ExtendedKey.deserialize() and throw an Exception if the bytes are not a valid xpub (including checksum).

Reason: it is better to concentrate xpub parsing in one location instead of many locations, making future maintenance easier.

Reply:
Changed to throw an exception in `ExtendedKey.deserialize`.